tests/integration/high_availability/high_availability_helpers_new.py

```diff
@@ -121,19 +107,23 @@ def get_app_units(juju: Juju, app_name: str) -> dict[str, UnitStatus]:
 
 def scale_app_units(juju: Juju, app_name: str, num_units: int) -> None:
     """Scale a given application to a number of units."""
-    app_units = get_app_units(juju, app_name)
+    app_units = list(get_app_units(juju, app_name))
     app_units_diff = len(app_units) - num_units
 
     scale_func = None
+    scale_func_args = []
+
     if app_units_diff > 0:
         scale_func = juju.remove_unit
+        scale_func_args = app_units[(-1 * app_units_diff) :]
     if app_units_diff < 0:
         scale_func = juju.add_unit
+        scale_func_args = [app_name] * (-1 * app_units_diff)
     if app_units_diff == 0:
         return
```

Comment:
```suggestion
    num_units_to_add = num_units - len(app_units)

    if num_units_to_add > 0:
        juju.add_unit(app_name, num_units=num_units_to_add)
    if num_units_to_add < 0:
        # e.g. if `num_units_to_add` is `-3`, last 3 items from `app_units`
        units_to_remove = app_units[num_units_to_add:]

        juju.remove_unit(*units_to_remove)
    if num_units_to_add == 0:
        return
```
& remove
https://github.com/canonical/mysql-operator/blob/29a8bf1997ebf815cfb41d6ccfa35a9b05de108e/tests/integration/high_availability/high_availability_helpers_new.py#L125-L126
(couldn't include in suggestion)